policies/redis/producer/streams.go

```diff
@@ -73,7 +73,36 @@ func (e eventStore) RemoveDataset(ctx context.Context, token string, dsID string
 }
 
 func (e eventStore) EditDataset(ctx context.Context, token string, ds policies.Dataset) (policies.Dataset, error) {
-	return e.svc.EditDataset(ctx, token, ds)
+	ds, err := e.svc.EditDataset(ctx, token, ds)
```

Comment:
If we just edit a valid dataset, to change the sink to another one, it shouldn't send a Redis event to Agent.
We need a way to check it before sending this event...